lib/ret/delay_stop_signal_handler.ex

```diff
@@ -0,0 +1,39 @@
+# This gen_event handler allows a temporary delay of handling
+# SIGTERM to shut down the application.
+#
+# Calling delay_stop will cease handling SIGTERM, then calling
+# allow_stop will resume it. If SIGTERM was seen in the interim,
+# the application will then stop.
+defmodule Ret.DelayStopSignalHandler do
+  @moduledoc false
+
+  def init(_), do: {:ok, %{saw_sigterm: false}}
+
+  def delay_stop do
+    :ok =
+      :gen_event.swap_sup_handler(
+        :erl_signal_server,
+        {:erl_signal_handler, []},
+        {Ret.DelayStopSignalHandler, []}
+      )
+  end
+
+  def allow_stop do
+    :gen_event.call(:erl_signal_server, Ret.DelayStopSignalHandler, {:allow_stop, self()})
+
+    :ok =
+      :gen_event.swap_sup_handler(
+        :erl_signal_server,
+        {Ret.SignalHandler, []},
```

Comment:
What is `Ret.SignalHandler`?

Reply:
😱 
typo. i tested this in production and it seemed to work because migrations were not stomped, but obviously was timing related. nice catch. will need to re-do all the integration tests.